File: rust/src/models/metered_entitlement_usage.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MeteredEntitlementUsage {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub consumed: Option<rust_decimal::Decimal>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub remaining: Option<rust_decimal::Decimal>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub reset_at: Option<String>,
}

impl MeteredEntitlementUsage {
    pub fn new() -> Self {
        Self {
            consumed: None,
            remaining: None,
            reset_at: None,
        }
    }
}

File: python/meteroid/models/metered_entitlement_usage.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime
from decimal import Decimal

from ..serialization import BaseModel


@dataclasses.dataclass
class MeteredEntitlementUsage(BaseModel):
    consumed: t.Optional[Decimal] = None

    remaining: t.Optional[Decimal] = None

    reset_at: t.Optional[datetime] = None
